File: NERP.NetworkEntities/Models/Cable.cs
using NERP.Common.Models;

namespace NERP.NetworkEntities.Models
{
    public class Cable : BaseNetworkEntity
    {
        public string ALocation { get; set; }
        public string BLocation { get; set; }
        public LineString Geometry { get; set; }
        public bool? IsUnderground { get; set; }
        public double? MeasuredLength { get; set; }
        public double? CalculatedLength { get; set; }
        public int? NoOfTubes { get; set; }
        public int? CorePerTube { get; set; }
        //public string Core_Access { get; set; }
        //public List<ColorSettings> TubeColorSettings { get; set; }
        //public List<ColorSettings> CoreColorSettings { get; set; }
        public Optics Optics { get; set; }
        public Installation Installation { get; set; }
        public Owner Owner { get; set; }
    }
}

File: NERP.Admin.Web1/ClientApp/src/app/entities/models/cable.ts
import { LineString } from '../../common/geo/geometry';
import { BaseNetworkEntity } from './baseNetworkEntity';
import { Installation } from './installation';
import { Optics } from './optics';
import { Owner } from './owner';

export class Cable extends BaseNetworkEntity {

    aLocation?: string;
    bLocation?: string;
    geometry: LineString;
    isUnderground?: boolean;
    measuredLength?: number;
    calculatedLength?: number;
    noOfTubes?: number;
    corePerTube?: number;
    optics?: Optics = new Optics();
    installation?: Installation = new Installation();
    owner?: Owner = new Owner();

    constructor(id?:string, code?: string, isUnderground?: boolean, noOfTubes?: number, corePerTube?: number, optics?: Optics) {
        super(id, code);
        this.isUnderground = isUnderground;
        this.noOfTubes = noOfTubes;
        this.corePerTube = corePerTube;
        this.optics = optics==null? new Optics(): optics;
    }
}

